ui/common/Input.tsx

```diff
@@ -1,39 +1,93 @@
-'use client'
-
-import RICIBs from 'react-individual-character-input-boxes'
-
-export default function Input({
-  amount,
-  onChange,
-}: {
-  amount: number
-  onChange: Function
-}) {
-  const handleChange = (value: string) => {
-    onChange(value)
-  }
-
-  return (
-    <RICIBs
-      amount={amount}
-      autoFocus
-      handleOutputString={handleChange}
-      inputProps={{
-        className: 'bg-transparent',
-        placeholder: '_',
-        style: {
-          fontSize: '20px',
-          width: '20px',
-          height: '20px',
-          margin: '0px',
-          borderRadius: '0px',
-          textAlign: 'center',
-          justifyContent: 'space-evenly',
-          outline: 'none',
-          fontFamily: 'var(--space-mono-font)',
-        },
-      }}
-      inputRegExp={/^[a-zA-Z0-9_.-]*$/}
-    />
-  )
-}
+'use client'
+
+import { useState } from 'react'
+
+interface UserInputProps {
+  onChange: Function
+  answer: string
+  hints?: boolean
+}
+
+export default function Input({ onChange, answer, hints }: UserInputProps) {
+  const [textAreaValue, setTextAreaValue] = useState('')
+  const [correctAns, setCorrectAns] = useState(false)
+
+  const handleKeyDown = (event: React.KeyboardEvent<HTMLTextAreaElement>) => {
+    if (event.key === ' ') {
+      event.preventDefault()
+    }
+  }
+
+  const blankSpace = () => {
+    let underscores = '_'.repeat(answer.length)
+    underscores = underscores.substring(0, answer.length - textAreaValue.length)
+    return (
+      <>
+        {Array.from({ length: answer.length }, (_, i) => {
+          if (i >= answer.length - underscores.length)
+            return (

```

Comment:
Another nitpick but let's always use curly braces, even for switches. Though its not incorrect to do it this way in this specific scenario I think curly braces help prevent issues with scoping, so it's a good practice to have.